### ISSUE TITLE
"version 2.0.2 doesnt work on firefox"

### ISSUE THREAD

**Issue by OnlyNandan** [tomer8007/whatsapp-web-incognito#147]
Installed v2.0.2. had to change manifest.txt as it gave me a  /background/service_worker" is not supported error. so i changed it to

 },
   "background": {
    "scripts": ["background.js"]
  },
  
  
the add on was uploaded properly. but when i open whatsapp i get nothing in that extension 
![image](https://github.com/tomer8007/whatsapp-web-incognito/assets/78302872/449bb4df-c615-4c00-b671-5ab4642aca8b)

**Comment by tomer8007** [comment]
Hi, 
I never designed this extension to work for Firefox, but first please try the 2.1.0 release I just published.
Also look for JS errors.

**Comment by OnlyNandan** [comment]
Oh noted. I tried the 2.1.0 release and it have me the same "background.service_worker is currently disabled" error

**Comment by tomer8007** [comment]
This sounds like a manifest v2/v3 issue. Are you aware of #38?

**Comment by OnlyNandan** [comment]
yep i tried #38 and i was able to intstall vs 2.0.1 on firefox which worked flawlessly . 

v 2.0.1
},
  "background": {
    "scripts": ["browser-polyfill.js", "background.js"]
  },
  
  
  and v 2.0.2
  
  "background":
   {
    "service_worker": "background.js"
   },
   "host_permissions": ["https://web.whatsapp.com/*"],
   "content_scripts": 
   [
  	{
  	
  there is a reference to service worker in  v 2.0.2 and v 2.1.0
  
  
![image](https://github.com/tomer8007/whatsapp-web-incognito/assets/78302872/309f2f37-12a6-43ad-aecf-e566579b7dd1)


other errors can be resolved with the solutions present in #38

**Comment by tomer8007** [comment]
So what is your error currently?

**Comment by OnlyNandan** [comment]
Main issue being that if i add v 2.0.1 to the ff then it get added and works without issue since it doesnt have background service worker. whereas the counterpart v 2.0.2 or 2.1.0 has background service worker which throws an error when added to ff. as ff doesnt support background service worker. and i was wondering if anyhow it would be possible to run the v2.0.2 upwards on ff.

**Comment by tomer8007** [comment]
The service worker is a manifest v3 thing I believe, so this makes sense. You could port it back to manifest v2 and convert the service worker to a background page just like you did. You could also figure out if Firefox supports manifest v3 (last time I checked it didn't).

**Comment by kamzone** [comment]
Hi. 
You can try migrating it to manifest v3 using [this guide](https://extensionworkshop.com/documentation/develop/manifest-v3-migration-guide/). It's pretty good.
It worked for different extensions. I'll try doing it, and if it will work I'll post an update.

**Comment by OnlyNandan** [comment]
Hey kamzone,
Thanks for the update ill try this out if i am able to get it to work ill update. i can get it to work please do the same.

**Comment by terrasaw** [comment]
Hi @OnlyNandan, did you manage to make it work?

**Comment by OnlyNandan** [comment]
Hey, 
Nope no luck as of now. only got the older versions running.

**Comment by bayazidbh** [comment]
For context - what does Firefox do differently or otherwise still not support that's making this remain a blocker for newer versions of the extension?

**Comment by OnlyNandan** [comment]
> For context - what does Firefox do differently or otherwise still not support that's making this remain a blocker for newer versions of the extension?
> 
> And what about version 2.0.1? Is that fine?

Its with regards to firefox not supporting manifest v3 yet

**Comment by bayazidbh** [comment]
@OnlyNandan is there something they haven't implemented yet, or something I'm missing, since [according to this](https://extensionworkshop.com/documentation/develop/manifest-v3-migration-guide/) they already have it available?
